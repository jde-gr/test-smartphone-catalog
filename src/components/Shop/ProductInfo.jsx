import PropTypes from 'prop-types';
import {
  Image,
  ImgContainer,
  Content,
  Details,
  Title,
  Price,
  SubTitle,
  Desc,
  Key,
} from './styles/ProductDetails.styles';

export const ProductInfo = ({
  brand,
  model,
  price,
  cpu,
  ram,
  os,
  displayResolution,
  displayType,
  displaySize,
  internalMemory,
  externalMemory,
  battery,
  primaryCamera,
  secondaryCmera,
  dimentions,
  weight,
}) => {
  return (
    <Details>
      <Title>
        {brand} {model}
      </Title>
      <Price>Precio: {price}€</Price>
      <SubTitle>Características:</SubTitle>
      <Desc>
        <Key>CPU:</Key> {cpu}
      </Desc>
      <Desc>
        <Key>RAM:</Key> {ram}
      </Desc>
      <Desc>
        <Key>Sistema Operativo:</Key> {os}
      </Desc>
      <Desc>
        <Key>Resolución de pantalla:</Key> {displayResolution}
      </Desc>
      <Desc>
        <Key>Tipo de pantalla:</Key> {displayType}
      </Desc>
      <Desc>
        <Key>Tamaño de pantalla:</Key> {displaySize}
      </Desc>
      <Desc>
        <Key>Memoria interna:</Key> {internalMemory}
      </Desc>
      <Desc>
        <Key>Memoria externa:</Key> {externalMemory}
      </Desc>
      <Desc>
        <Key>Batería:</Key> {battery}
      </Desc>
      <Desc>
        <Key>Cámara principal:</Key> {primaryCamera}
      </Desc>
      <Desc>
        <Key>Cámara secundaria:</Key> {secondaryCmera}
      </Desc>
      <Desc>
        <Key>Dimensiones:</Key> {dimentions}
      </Desc>
      <Desc>
        <Key>Peso:</Key> {weight}g.
      </Desc>
    </Details>
  );
};

ProductInfo.propTypes = {
  imgUrl: PropTypes.string,
  brand: PropTypes.string,
  model: PropTypes.string,
  price: PropTypes.string,
  cpu: PropTypes.string,
  ram: PropTypes.string,
  os: PropTypes.string,
  displayResolution: PropTypes.string,
  displayType: PropTypes.string,
  displaySize: PropTypes.string,
  internalMemory: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  externalMemory: PropTypes.string,
  battery: PropTypes.string,
  primaryCamera: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  secondaryCmera: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  dimentions: PropTypes.string,
  weight: PropTypes.string,
};

export default ProductInfo;
