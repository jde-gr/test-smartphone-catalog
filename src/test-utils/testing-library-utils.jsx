import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store';

const queryClient = new QueryClient();

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          {children}
        </QueryClientProvider>
      </Provider>
    </BrowserRouter>
  );
};
const renderWithContext = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { renderWithContext as render };
