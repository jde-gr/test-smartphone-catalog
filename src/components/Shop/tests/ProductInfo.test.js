import { render } from '../../../test-utils/testing-library-utils';
import ProductInfo from '../ProductInfo';

test('renders correctly', () => {
  const { container } = render(<ProductInfo />);
  expect(container).toMatchSnapshot();
});
