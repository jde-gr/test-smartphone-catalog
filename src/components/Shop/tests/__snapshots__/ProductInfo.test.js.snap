// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-strwe"
  >
    <h2
      class="css-1hfgpi1"
    >
       
    </h2>
    <h5
      class="css-19fz2k3"
    >
      Precio: 
      €
    </h5>
    <h4
      class="css-zvcg02"
    >
      Características:
    </h4>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        CPU:
      </span>
       
    </p>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        RAM:
      </span>
       
    </p>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        Sistema Operativo:
      </span>
       
    </p>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        Resolución de pantalla:
      </span>
       
    </p>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        Tipo de pantalla:
      </span>
       
    </p>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        Tamaño de pantalla:
      </span>
       
    </p>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        Memoria interna:
      </span>
       
    </p>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        Memoria externa:
      </span>
       
    </p>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        Batería:
      </span>
       
    </p>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        Cámara principal:
      </span>
       
    </p>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        Cámara secundaria:
      </span>
       
    </p>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        Dimensiones:
      </span>
       
    </p>
    <p
      class="css-usxmki"
    >
      <span
        class="css-1ezgv1"
      >
        Peso:
      </span>
       
      g.
    </p>
  </div>
</div>
`;
